### ISSUE TITLE
aggregators are reported every batch

### ISSUE THREAD

**Issue by gideonite** [ai2cm/ace#281]
Ah I think I understand more what Brian was saying now that the initial issue of needing a different aggregator per epoch is fixed. I think this is OK for this PR, but I would suggest bringing this in line with what we do in core, which is:
- log the batch loss (and only the loss) only every N batches
- log the train aggregator metrics at the end of the epoch

Currently you need to do a wandb call and an aggregator collection at the end of each batch, which I expect will significantly slow down your training. If you wanted to fix this performance issue quickly for this PR, it would be fine to pull this wandb log call to the end of the training loop, the way you do in valid_one_epoch. The downside is you won't get per-batch loss values recorded (but that's easy to add in its own PR).

To be clear, this is the correct pattern for the aggregator (collecting across all batches in the epoch), it's just that we don't need to see its metrics until the epoch is finished). We don't want to, in fact, because it would make it very hard to compare results between epochs.

Non-blocking, but if you don't do anything please open an issue.

_Originally posted by @mcgibbon in https://github.com/ai2cm/full-model/pull/529#discussion_r1488102277_
